eclipse-che/che-operator
# feat: don't take control over DWO which is not managed by Che

### What does this PR do?
It's really annoying when you do `make install` from DWO repo, and expect your custom DWO be installed, but then Che operator takes control over DWO.
There is a weird way to workaround it - cluster DWO CSV, but I believe the proper solution is not to take control over DWO which is not managed by Che.

Initially, it was supposed to be simple PR, but then I noticed a few bugs and inaccuracies, so it also brings refactoring to the whole devworkspace package.


### Screenshot/screencast of this PR



### What issues does this PR fix or reference?
N/A


### How to test this PR?
unit tests pretty everything, but for manual test you may:
1. Checkout https://github.com/devfile/devworkspace-operator and execute make install from there.
2. Deploy the with operator installer.
3. Check that DWO from 1 is not overridden.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
